mantle/platform/api/aws/ec2.go

```diff
@@ -108,77 +108,97 @@ func (a *API) CreateInstances(name, keyname, userdata string, count uint64, minD
-		SubnetId:            &subnetId,
-		UserData:            ud,
-		BlockDeviceMappings: rootBlockDev,
-		TagSpecifications: []*ec2.TagSpecification{
-			&ec2.TagSpecification{
-				ResourceType: aws.String(ec2.ResourceTypeInstance),
-				Tags: []*ec2.Tag{
-					&ec2.Tag{
-						Key:   aws.String("Name"),
-						Value: aws.String(name),
-					},
-					&ec2.Tag{
-						Key:   aws.String("CreatedBy"),
-						Value: aws.String("mantle"),
+		key := &keyname
+		if keyname == "" {
+			key = nil
+		}
+
+		var rootBlockDev []*ec2.BlockDeviceMapping
+		if minDiskSize > 0 {
+			rootBlockDev = append(rootBlockDev, &ec2.BlockDeviceMapping{
+				DeviceName: aws.String("/dev/xvda"),
+				Ebs: &ec2.EbsBlockDevice{
+					VolumeSize: &minDiskSize,
+				},
+			})
+		}
+		inst := ec2.RunInstancesInput{
+			ImageId:             &a.opts.AMI,
+			MinCount:            &cnt,
+			MaxCount:            &cnt,
+			KeyName:             key,
+			InstanceType:        &a.opts.InstanceType,
+			SecurityGroupIds:    []*string{&sgId},
+			SubnetId:            &subnetId,
+			UserData:            ud,
+			BlockDeviceMappings: rootBlockDev,
+			TagSpecifications: []*ec2.TagSpecification{
+				&ec2.TagSpecification{
+					ResourceType: aws.String(ec2.ResourceTypeInstance),
+					Tags: []*ec2.Tag{
+						&ec2.Tag{
+							Key:   aws.String("Name"),
+							Value: aws.String(name),
+						},
+						&ec2.Tag{
+							Key:   aws.String("CreatedBy"),
+							Value: aws.String("mantle"),
+						},
 					},
 				},
 			},
-		},
-	}
-	if useInstanceProfile {
-		inst.IamInstanceProfile = &ec2.IamInstanceProfileSpecification{
-			Name: &a.opts.IAMInstanceProfile,
 		}
-	}
+		if useInstanceProfile {
+			inst.IamInstanceProfile = &ec2.IamInstanceProfileSpecification{
+				Name: &a.opts.IAMInstanceProfile,
+			}
+		}
 
-	var reservations *ec2.Reservation
-	err = util.RetryConditional(5, 5*time.Second, func(err error) bool {
-		// due to AWS' eventual consistency despite ensuring that the IAM Instance
-		// Profile has been created it may not be available to ec2 yet.
-		if awsErr, ok := err.(awserr.Error); ok && (awsErr.Code() == "InvalidParameterValue" && strings.Contains(awsErr.Message(), "iamInstanceProfile.name")) {
-			return true
+		err = util.RetryConditional(5, 5*time.Second, func(err error) bool {
+			// due to AWS' eventual consistency despite ensuring that the IAM Instance
+			// Profile has been created it may not be available to ec2 yet.
+			if awsErr, ok := err.(awserr.Error); ok && (awsErr.Code() == "InvalidParameterValue" && strings.Contains(awsErr.Message(), "iamInstanceProfile.name")) {
+				return true
+			}
+			return false
+		}, func() error {
+			var ierr error
+			reservations, ierr = a.ec2.RunInstances(&inst)
+			return ierr
+		})
+		if err == nil {
+			// Successfully started our instance in the requested zone. Break out of the loop
+			break
+		}
+		if err != nil {
+			// Handle InsufficientInstanceCapacity error by looking for instances in other zones
```

Comment:
I think less is more here. Maybe something like:

```suggestion
			// Handle InsufficientInstanceCapacity error specifically
```